jerry-core/ecma/builtin-objects/ecma-builtin-regexp-prototype.c

```diff
@@ -627,76 +501,112 @@ inline bool JERRY_ATTR_ALWAYS_INLINE
 ecma_builtin_is_regexp_exec (ecma_extended_object_t *obj_p)
 {
   return (ecma_get_object_is_builtin ((ecma_object_t *) obj_p)
-          && obj_p->u.built_in.routine_id == ECMA_ROUTINE_LIT_MAGIC_STRING_EXECecma_builtin_regexp_prototype_exec);
+          && obj_p->u.built_in.routine_id == ECMA_REGEXP_PROTOTYPE_ROUTINE_EXEC);
 } /* ecma_builtin_is_regexp_exec */
+#endif /* ENABLED (JERRY_ESNEXT) */
 
 /**
- * The RegExp.prototype object's '@@replace' routine
- *
- * See also:
- *          ECMA-262 v6.0, 21.2.5.8
+ * Dispatcher of the Regexp built-in's routines
  *
  * @return ecma value
  *         Returned value must be freed with ecma_free_value.
  */
-static ecma_value_t
-ecma_builtin_regexp_prototype_symbol_replace (ecma_value_t this_arg, /**< this argument */
-                                              ecma_value_t string_arg, /**< source string */
-                                              ecma_value_t replace_arg) /**< replace string */
+ecma_value_t
+ecma_builtin_regexp_prototype_dispatch_routine (uint16_t builtin_routine_id, /**< built-in wide routine
+                                                                              *   identifier */
+                                                ecma_value_t this_arg, /**< 'this' argument value */
+                                                const ecma_value_t arguments_list_p[], /**< list of arguments
+                                                                                        *   passed to routine */
+                                                ecma_length_t arguments_number) /**< length of arguments' list */
 {
-  return ecma_regexp_replace_helper (this_arg, string_arg, replace_arg);
-} /* ecma_builtin_regexp_prototype_symbol_replace */
+  JERRY_UNUSED (arguments_number);
 
-/**
- * The RegExp.prototype object's '@@search' routine
- *
- * See also:
- *          ECMA-262 v6.0, 21.2.5.9
- *
- * @return ecma value
- *         Returned value must be freed with ecma_free_value.
- */
-static ecma_value_t
-ecma_builtin_regexp_prototype_symbol_search (ecma_value_t this_arg, /**< this argument */
-                                             ecma_value_t string_arg) /**< string argument */
-{
-  return ecma_regexp_search_helper (this_arg, string_arg);
-} /* ecma_builtin_regexp_prototype_symbol_search */
+#if ENABLED (JERRY_BUILTIN_ANNEXB)
+  if (builtin_routine_id == ECMA_REGEXP_PROTOTYPE_ROUTINE_COMPILE)
+  {
+    return ecma_builtin_regexp_prototype_compile (this_arg, arguments_list_p[0], arguments_list_p[1]);
```

Comment:
Why do we need to handle this separately? This routine also requires the this value to be a RegExp object, that could be handled with the others, and leave only the Regexp.prototype check in the routine.